src/main/ruby/truffleruby/core/truffle/io_operations.rb

```diff
@@ -245,7 +245,7 @@ def self.poll(io, event_mask, timeout)
                 # Update timeout
                 now = Process.clock_gettime(Process::CLOCK_MONOTONIC, :millisecond)
                 if now >= deadline
-                  false # timeout
+                  0 # timeout
```

Comment:
Doesn't it break the logic above in

```ruby
        while timeout_ms > 2147483647 # INT_MAX
          timeout_ms -= 2147483000
          ret = poll(io, event_mask, 2147483)
          return ret unless Primitive.false?(ret)
        end
```

Reply:
Right, looked at all uses except the closest one. Fixing it.
Also - that's iterating on a 24 day long wait :O